docs/odh/oci-model-storage.md

```diff
@@ -8,15 +8,228 @@ which also explains how to deploy models from OCI images.
+to demonstrate how to deploy the Granite model stored in an OCI image.
+
+### Storing the Granite model in an OCI image
+
+To download the Granite model, the [`huggingface-cli download` command](https://huggingface.co/docs/huggingface_hub/guides/cli#huggingface-cli-download) can be used.
+The process is as follows:
+* Install the huggingface CLI
+* Use huggingface CLI to download the model
+* Create the final OCI using the downloaded model
+
+This process is implemented in the following multi-stage container build. Create a file named
+`Containerfile` with the following contents:
+```Dockerfile
+##### Stage 1: Download the model
+FROM registry.access.redhat.com/ubi9/python-312:latest as downloader
+
+# Install huggingface-cli
+RUN pip install "huggingface_hub[cli]"
+
+# Download the model
+ARG repo_id
+RUN mkdir models && huggingface-cli download --quiet --max-workers 2 --local-dir ./models $repo_id
+
+##### Stage 2: Build the final OCI model container
+FROM registry.access.redhat.com/ubi9/ubi-micro:latest as model
+
+# Copy from the download stage
+COPY --from=downloader --chown=1001:0 /opt/app-root/src/models /models
+
+# Set proper privileges for KServe
+RUN chmod -R a=rX /models
+
+# Use non-root user as default
+USER 1001
+```
+
+> [!TIP]
+> This Containerfile should be generic enough to download and containerize any
+> model from Hugging Face. However, it has only been tested with the Granite model.
+> Feel free to try it with any other model that can work with the vLLM server.
+
+Notice that model files are copied into `/models` inside the final container. KServe
+expects this path to exist in the OCI image and also expects the model files to
+be inside it.
+
+Also, notice that `ubi9-micro` is used as a base container of the final image.
+Empty images, like `scratch` cannot be used, because KServe needs to configure the model image
+with a command to keep it alive and ensure the model files remain available in
+the pod. Thus, it is required to use a base image that provides a shell.
+
+Finally, notice that ownership of the copied model files is changed to the `root`
+group, and also read permissions are granted. This is important, because OpenShift
+runs containers with a random user ID and with the `root` group ID. The adjustment
+of the group and the privileges on the model files ensures that the model server
+can access them.
+
+Create the OCI container image of the Granite model using Podman, and upload it to
+a registry. For example, using Quay as the registry:
+```shell
+podman build --format=oci --squash \
+  --build-arg repo_id=ibm-granite/granite-3.0-2b-instruct \
+  -t quay.io/<user_name>/<repository_name>:<tag_name> .
+
+podman push quay.io/<user_name>/<repository_name>:<tag_name>
+```
+
+It is important to use the `--squash` flag to prevent the final image having
+the double size of the model.
+
+> [!TIP]
+> When uploading your container image, if your repository is private, ensure you
+> are authenticated to the registry.
```

Comment:
If the repo is private, what happens when KServer tries to fetch it?

Reply:
It would fail if you don't configure the credentials. This is explained in the very last section of the file (but the diff is not showing it, since it is unchanged).